test/math.jl

```diff
@@ -842,24 +842,103 @@ end
+        @test sinh(T(1000)) === T(Inf)
+        @test sinh(-T(1000)) === -T(Inf)
+        @test isnan_type(T, sinh(T(NaN)))
+    end
+    for x in (pcnfloat(2.0^-28)..., pcnfloat(22.0)..., pcnfloat(709.7822265633563)...)
+        @test sinh(x) ≈ sinh(big(x)) rtol=eps(Float64)
+        @test sinh(-x) ≈ sinh(big(-x)) rtol=eps(Float64)
+    end
+    for x in (pcnfloat(2f-12)..., pcnfloat(9f0)..., pcnfloat(88.72283f0)...)
+        @test sinh(x) ≈ sinh(big(x)) rtol=eps(Float32)
+        @test sinh(-x) ≈ sinh(big(-x)) rtol=eps(Float32)
+    end
+end
 
-@testset "acos #23283" begin
+@testset "cosh" begin
     for T in (Float32, Float64)
-        @test acos(zero(T)) === T(pi)/2
-        @test acos(-zero(T)) === T(pi)/2
-        @test acos(nextfloat(zero(T))) === T(pi)/2
-        @test acos(prevfloat(zero(T))) === T(pi)/2
-        @test acos(one(T)) === T(0.0)
-        @test acos(-one(T)) === T(pi)
-        for x in (0.45, 0.6, 0.98)
-            by = acos(big(T(x)))
-            @test T((acos(T(x)) - by))/eps(abs(T(by))) <= 1
-            bym = acos(big(T(-x)))
-            @test T(abs(acos(T(-x)) - bym))/eps(abs(T(bym))) <= 1
+        @test cosh(zero(T)) === one(T)
+        @test cosh(-zero(T)) === one(T)
+        @test cosh(nextfloat(zero(T))) === one(T)
+        @test cosh(prevfloat(zero(T))) === one(T)
+        @test cosh(T(1000)) === T(Inf)
+        @test cosh(-T(1000)) === T(Inf)
+        @test isnan_type(T, cosh(T(NaN)))
+    end
+    for x in (pcnfloat(2.7755602085408512e-17)..., pcnfloat(22.0)..., pcnfloat(709.7822265633563)...)
+        @test cosh(x) ≈ cosh(big(x)) rtol=eps(Float64)
+        @test cosh(-x) ≈ cosh(big(-x)) rtol=eps(Float64)
+    end
+    for x in (pcnfloat(0.00024414062f0)..., pcnfloat(9f0)..., pcnfloat(88.72283f0)...)
+        @test cosh(x) ≈ cosh(big(x)) rtol=eps(Float32)
+        @test cosh(-x) ≈ cosh(big(-x)) rtol=eps(Float32)
+    end
+end
+
+@testset "tanh" begin
+    for T in (Float32, Float64)
+        @test tanh(zero(T)) === zero(T)
+        @test tanh(-zero(T)) === -zero(T)
+        @test tanh(nextfloat(zero(T))) === nextfloat(zero(T))
+        @test tanh(prevfloat(zero(T))) === prevfloat(zero(T))
+        @test tanh(T(1000)) === one(T)
+        @test tanh(-T(1000)) === -one(T)
+        @test isnan_type(T, tanh(T(NaN)))
+        for x in Iterators.flatten(pcnfloat.(T == Float64 ? [2.0^-28,1.0,22.0] : [2f0^-12,1f0,9f0]))
+            @test tanh(x) ≈ tanh(big(x)) rtol=eps(T)
+            @test tanh(-x) ≈ tanh(big(-x)) rtol=eps(T)
+        end
+    end
+end
+
+@testset "asinh" begin
+    for T in (Float32, Float64)
+        @test asinh(zero(T)) === zero(T)
+        @test asinh(-zero(T)) === -zero(T)
+        @test asinh(nextfloat(zero(T))) === nextfloat(zero(T))
+        @test asinh(prevfloat(zero(T))) === prevfloat(zero(T))
+        @test isnan_type(T, asinh(T(NaN)))
+        for x in Iterators.flatten(pcnfloat.(T == Float64 ? [2.0^-28,2.0,2.0^28] : [2f0^-12,2f0,22f28]))
```

Comment:
I'm confused about this conditional.  Looking back at the `asinh` code, it seems like it uses `2^-28` and `2^28` for both single and double precision.     `2^-12` is only used for `tanh`.

Again, this would be clarified by using `TANH_SMALL_X(T)` etcetera here for precision-dependent thresholds.

Reply:
This is a good idea. Originally it was unsure if I there were going to be tests in separate commits before the actual change (just to make sure nothing changed), but I guess this will be squashed anyway, so might as well just use the thresholds from the actual code.